### PR TITLE
Enable single target builds

### PR DESCRIPTION
**Pull request by dansiegel**
GitHub Issue (If applicable): closes #

- Replaces #141

## PR Type

What kind of change does this PR introduce?

- Feature

## What is the current behavior?

The projects core "shared" project and the mobile project will multi-target as necessary.

## What is the new behavior?

Developers may now optionally uncomment an import line in the Directory.Build.props and rename the sample file to single target. This is an opt in feature. We additionally have added the official VisualStudio .gitignore with the addition of a line to ignore the solution-config.props so that if a developer chooses to use this they will not accidently check the file into source control.

It will be up to the developer to create any solution filters that they may want. The linked docs explain this process in more detail.

### CODE REVIEW

File: src/Uno.Templates/Uno.Templates.csproj [hunk 1]
```diff
@@ -47,6 +47,15 @@
 
 		<RemoveDir Directories="$(TemplateContentFolder)" />
 
+		<!-- Ensure we keep the .gitignore up to date with the latest changes to the official VisualStudio.gitignore -->
+		<DownloadFile
```
Comment (nickrandolph) [inline src/Uno.Templates/Uno.Templates.csproj:51]: Can we move this to a different PR  - this is something that can go forward regardless

Comment (dansiegel) [reply src/Uno.Templates/Uno.Templates.csproj:51]: now that we're moving this to ready for review this should be fine to leave here

**Review by nickrandolph — APPROVED** [review]
